Code/Tools/AssetProcessor/native/InternalBuilders/SettingsRegistryBuilder.cpp

```diff
@@ -279,18 +285,47 @@ namespace AssetProcessor
 
                     for (const auto& settingsKey : settingsToCopy)
                     {
-                        AZ::SettingsRegistryInterface::FixedValueString settingsValue;
+                        FixedValueString settingsValue;
                         [[maybe_unused]] bool settingsCopied = settingsRegistry->Get(settingsValue, settingsKey)
                             && registry.Set(settingsKey, settingsValue);
                         AZ_Warning("Settings Registry Builder", settingsCopied, "Unable to copy setting %s from AssetProcessor settings registry"
                             " to local settings registry", settingsKey.c_str());
                     }
+
+                    // Read the AssetProcessor loaded Gem Information from the global Registry
+                    AZStd::vector<AzFramework::GemInfo> gemInfos;
+                    size_t pathIndex{};
+                    if (AzFramework::GetGemsInfo(gemInfos, *settingsRegistry))
+                    {
+                        AZStd::vector<AZ::IO::PathView> sourcePaths;
```

Comment:
Wouldn't a set be more efficient? It would remove the need to repeatedly search the entire list.

Reply:
I went with vector instead of a set to maintain the order of merging that the AssetProcessors uses for the Game Launcher settings.
That being said since the Gems Registries should act independent from each other, the order shouldn't make a difference